www/source/docs/reference.html.md.erb

```diff
@@ -23,24 +24,1480 @@ This section will cover as many of the modules and APIs that Chef Habitat is bui
+```
+
+Example pointing to specific file in <code>core/nginx</code> package on disk:
+
+```handlebars
+{{pkgPathFor "core/nginx"}}/config/fastcgi.conf
+```
+
+**eachAlive**
+: Iterates over a collection of members and renders the template for members that are marked alive.
+
+```handlebars
+{{~#eachAlive bind.backend.members as |member|}}
+server ip {{member.sys.ip}}:{{member.cfg.port}}
+{{~/eachAlive}}
+```
+
+**toJson**
+: To output configuration data as JSON, you can use the `toJson` helper.
+
+Given a default.toml that looks like:
+
+```toml
+[web]
+
+[[servers]]
+host = "host-1"
+port = 4545
+
+[[servers]]
+host = "host-2"
+port = 3434
+```
+
+and a template:
+
+```handlebars
+{{toJson cfg.web}}
+```
+
+when rendered, it will look like:
+
+```json
+{
+  "servers": [
+    {
+      "host": "host-1",
+      "port": 4545
+    },
+    {
+      "host": "host-2",
+      "port": 3434
+    }
+  ]
+}
+```
+
+This can be useful if you have a configuration file that is in JSON format and
+has the same structure as your TOML configuration data.
+
+**toToml**
+: The `toToml` helper can be used to output TOML.
+
+Given a default.toml that looks like:
+
+```toml default.toml
+[web]
+
+port = 80
+```
+
+and a template:
+
+```handlebars
+{{toToml cfg.web}}
+```
+
+when rendered, it will look like:
+
+```toml
+port = 80
+```
+
+This can be useful if you have an app that uses TOML as its configuration file
+format, but may have not been designed for Chef Habitat, and you only need certain
+parts of the configuration data in the rendered TOML file.
+
+**toYaml**
+: The `toYaml` helper can be used to output [YAML](http://yaml.org/).
+
+Given a default.toml that looks like:
+
+```toml default.toml
+[web]
+
+port = 80
+```
+
+and a template:
+
+```handlebars
+{{toYaml cfg}}
+```
+
+when rendered, it will look like:
+
+```yaml
 ---
-<%= partial "/partials/global/helpers"%>
+web:
+  port: 80
+```
+
+The helper outputs a YAML document (with a line beginning with `---`), so it must be used to create complete documents: you cannot insert a section of YAML into an existing YAML document with this helper.
+
+**strJoin**
+: The `join` helper can be used to create a string with the variables in a list with a separator specified by you. For example, where `list: ["foo", "bar", "baz"]`, `{{strJoin list ","}}` would return `"foo,bar,baz"`.
+
+You cannot join an object (e.g. `{{strJoin web}}`), but you could join the variables in an object (e.g. `{{strJoin web.list "/"}}`).
+
+
+**strConcat**
+: The `concat` helper can be used to connect multiple strings into one string without a separator. For example, `{{strConcat "foo" "bar" "baz"}}` would return `"foobarbaz"`.\
+
+You cannot concatenate an object (e.g. `{{strConcat web}}`), but you could concatenate the variables in an object (e.g. `{{strConcat web.list}}`).
+
 ---
-<%= partial "/partials/docs/reference-template-data"%>
+## <a name="template-data" id="template-data" data-magellan-target="template-data">Template Data</a>
```

Comment:
I deleted these lines from the partials file, because they don't work and appear in the text:

```
 ARGV tmp/render_context_schema.json	
<!-- This is a generated file, do not edit it directly. See https://github.com/habitat-sh/habitat/blob/master/www/scripts/generate-template-reference.js -->
```

https://www.habitat.sh/docs/reference/#concat-helper

Reply:
I'm not really sure what the intent was here. It looks like diagnostic logging that wasn't supposed to end up in the generated output. It was added in https://github.com/habitat-sh/habitat/pull/6427, which @raskchanky reviewed, so he may have more context.

Reply:
The `ARGV` line appears to be a bug in how the CLI documentation script generates its output.  That's worth reviewing.  The HTML comment is generated automatically by the node.js script that generates the template reference docs.  This change to eliminate all the partials breaks that script, which overwrote a specific partial every time it ran.

Either the script will need to be adjusted to account for this new layout, which is likely a fairly complicated change, or we should make an exception in this case and keep this partial.

Reply:
This ARGV line exists in the live docs: 
<img width="1116" alt="Screen Shot 2019-08-15 at 4 05 24 PM" src="https://user-images.githubusercontent.com/4400151/63132601-8894a580-bf76-11e9-84ea-2e67d4edacfb.png">

Reply:
The template-data partial is the only auto-generated part of the reference docs and it is infrequently updated.